power_grid_model_c/power_grid_model/include/power_grid_model/topology.hpp

```diff
@@ -530,6 +530,7 @@ class Topology {
                 coupling[topo_comp_i] = Idx2D{topo_idx, new_math_comp_i};
             }
         }
+        (void)std::forward<GetMathTopoComponent>(get_component_topo);
```

Comment:
why is this happening?

Reply:
this function captures `func` by perfect forwarding. It is not allowed by the standard to not forward a forwarded argument, or move a moved argument.

Reply:
See https://github.com/PowerGridModel/power-grid-model/pull/882#discussion_r1938948404.

The argument `get_component_topo` should be passed by value as lambdas.